### PR TITLE
redeem page

### PR DESCRIPTION
**Pull request by aeither**

### CODE REVIEW

File: webapp/src/app/mytokens/page.tsx [hunk 1]
```diff
@@ -5,10 +5,10 @@ perPage: 10
 */
 "use client";
 
-import { use, useCallback, useEffect, useState } from "react";
-import { useContractWrite, usePrepareContractWrite, useContractRead, useAccount, useContractInfiniteReads, paginatedIndexesConfig } from "wagmi";
-import { astaverdeContractConfig, usdcContractConfig } from "../../lib/contracts";
+import { Dispatch, SetStateAction, use, useCallback, useEffect, useState } from "react";
```
Comment (orpheuslummis) [inline webapp/src/app/mytokens/page.tsx:8]: automatic formatting please

File: webapp/src/app/mytokens/page.tsx [hunk 2]
```diff
@@ -107,16 +109,22 @@ function BatchRedeemCard({batch}:{batch: Batch}) {
...
 
-  const filteredArray = useCallback(() => {
+  const ownerTokens = useCallback(() => {
     if(ownedIndex) {
       return batch.token_ids.filter((_, index) => +ownedIndex[index].toString() === 1);
     }
   }, [batch.token_ids, ownedIndex]);
-  console.log("🚀 ~ file: page.tsx:117 ~ filteredArray ~ filteredArray:", filteredArray())
+ 
+  const { config } = usePrepareContractWrite({
```
Comment (orpheuslummis) [inline webapp/src/app/mytokens/page.tsx:119]: note: somehow `redeemTokens` still gets triggered on load, with no args, and reverts.

**Review by orpheuslummis — APPROVED** [review]
Approving as WIP

File: webapp/src/app/mytokens/page.tsx (hunk 2 above)
Comment (aeither) [reply webapp/src/app/mytokens/page.tsx:132]: updated conditions

File: webapp/src/app/mytokens/page.tsx (hunk 1 above)
Comment (aeither) [reply webapp/src/app/mytokens/page.tsx]: used prettier

**Pull request merged**